paxoskv/impl.go

```diff
@@ -160,11 +160,9 @@ func (p *Proposer) rpcToAll(acceptorIds []int64, action string) []*Acceptor {
 			log.Fatalf("did not connect: %v", err)
 		}
 
-		defer conn.Close()
 		c := NewPaxosKVClient(conn)
 
 		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-		defer cancel()
```

Comment:
Is there any issue with `defer`?

Is the manually close just for reclaiming resource sooner? If it is, I'd prefer installing a closer as soon as creating it. Since it is just a tutorial project, not for production use.

Reply:
in the for loop, use defer, the `conn` will be 变量逃逸..

Reply:
It does not seem 变量逃逸了:

```go
package main

import "fmt"

type A struct {
	a int
}

func (v *A) leave() {
	fmt.Println("a=", v.a)
}

func main() {
	for i := 0; i < 3; i++ {
		x := A{a: i}
		defer x.leave()
	}

}
```

`go run x.go` will prints 3 lines.